Move the search bar control into the site-title.

StevenDufresne opened the pull request:
This PR is an experiment related to #40. Using the pattern directory & learn as the model, move the search bar into the site title.

**Example**
- [https://wordpress.org/patterns/categories/gallery/](https://wordpress.org/patterns/categories/gallery/)
- [https://learn.wordpress.org/workshops/](https://learn.wordpress.org/workshops/)

## Screenshots

### Function/Class/Hook
| Before |  After |
| --- | --- | 
| ![](https://d.pr/i/ytaa8c.png)  |  ![](https://d.pr/i/XpXC4c.png) |
|   | ![](https://d.pr/i/hM5hkB.png) |

### `is_page( 'reference' )`

I removed everything below the breadcrumb and above the 2 boxes since I didn't really know what to do with the layout and I'm not convinced they are used much. 

| Before |  After |
| --- | --- | 
| ![](https://d.pr/i/w6M7aO.png) |  ![](https://d.pr/i/WBYTgu.png) |
|  | ![](https://d.pr/i/m3zoDl.png) |

### `is_search()`
| Before |  After |
| --- | --- | 
| ![](https://d.pr/i/5h0VdP.png)  |  ![](https://d.pr/i/0Bs8Zg.png) |

### `is_handbook && is_search()`
| Before |  After |
| --- | --- | 
| ![](https://d.pr/i/IljlW8.png)  | ![](https://d.pr/i/CHjj42.png)  |

outdoor2kode commented:
This looks great!

Done testing on Mac/Chrome & Firefox with different screen sizes. 
Some thoughts:
1. The filter is removed from the API detail page, I think it's ok to me as it's still kept on the reference home page and I personally didn't use that filter that much. But not sure if it's a convenient tool for a veteran developer?

2. I saw there's a tiny space under the search icon and there's no outline style when clicking on it, perhaps we can give it some styles just like the one in the pattern directory?
![image](https://user-images.githubusercontent.com/18050944/172439418-b501944d-94d4-4155-a0b9-0adde2946d3b.png)
![image](https://user-images.githubusercontent.com/18050944/172439741-2c0d4522-e877-4aad-8154-e5a2eca91ff3.png)

3. It would be great if we can change the cursor to pointer for these two icons
![image](https://user-images.githubusercontent.com/18050944/172440586-1f41626e-4c74-44b5-ab0e-07e8dd0e527c.png)

4. I seem not able to see the search bar on the searching page like your screenshot shows
![Screen Capture on 2022-06-08 at 01-09-25](https://user-images.githubusercontent.com/18050944/172442118-da106062-b4ca-411d-b19e-ded93092e3af.gif)
5. I noted that if I'd like to get rid of the search suggestion block, I need to use the mouse to click elsewhere on the page instead of just pressing the ESC button. If so, users need to select the filter options first before typing in the search or they must use a mouse. I think in the original UX, the suggestion block can be gotten rid of by pressing ESC.
![image](https://user-images.githubusercontent.com/18050944/172442817-c8dfb6f2-30ac-421c-9374-bec14e7b960d.png)
    5a. Also tested on Firefox, and the suggestion block can be closed by pressing ESC, but it has a weird behavior that after closing it, it will pop up again.
![Screen Capture on 2022-06-08 at 01-32-08](https://user-images.githubusercontent.com/18050944/172446983-e2e55726-9d10-4d45-a73c-e15bb026dc39.gif)

6. Curious about what "This needs some clean up" means in the Reference Home screenshot, does it mean that you expect some new style design for it?

7. Commit message typepo: Move `searhc` into site title

StevenDufresne commented:
Thanks for the detailed review! I should have been more clear. The code needs cleaning up and some of the UI interactions/layouts are still a bit rough. 

I am interested to get this approach validated since we don't have an agreed-on path forward on the search UX. Do you think this is a good idea (moving the search bar)? Is it an improvement?

outdoor2kode commented:
To me, it's an improvement and I'm probably more into this new layout for 3 reasons. 

1. It's more consistent with the design of other WP sites.
2. I like it I don't have to toggle on the search bar every time, it's an extra step as far as I'm concerned.
3. I like the new UI, it's more obvious and looks nicer by putting it in the title header.

StevenDufresne commented:
@renintw Ok, cycling back to your feedback. I think I addressed most of them except:

>I noted that if I'd like to get rid of the search suggestion block, I need to use the mouse to click elsewhere on the page instead of just pressing the ESC button. If so, users need to select the filter options first before typing in the search or they must use a mouse. I think in the original UX, the suggestion block can be gotten rid of by pressing ESC.

This is an external plugin and I suspect this functionality is already _broken_. We can address it in a later PR.

ryelle commented on source/wp-content/themes/wporg-developer/header.php:
```diff
@@ -24,33 +33,27 @@
+				<?php get_search_form(); ?>
+			<?php endif; ?>
+
+			<nav id="site-navigation" class="main-navigation" role="navigation">
+				<button class="menu-toggle dashicons dashicons-arrow-down-alt2" aria-controls="primary-menu" aria-expanded="false" aria-label="<?php esc_attr_e( 'Primary Menu', 'wporg' ); ?>"></button>
+				<?php
+				$active_menu = is_post_type_archive( 'command' ) || is_singular( 'command' ) ? 'devhub-cli-menu' : 'devhub-menu';
+				wp_nav_menu( array(
+					'theme_location'  => $active_menu,
+					'container_class' => 'menu-container',
+					'container_id'    => 'primary-menu',
+				) ); ?>
+			</nav>
```
If I add items to this menu, they show up kind of unexpectedly:

<img width="995" alt="Screen Shot 2022-06-09 at 3 24 04 PM" src="https://user-images.githubusercontent.com/541093/172928162-44404a9c-0602-4dc7-a2a8-e81809a8579a.png">

It looks like this menu is empty on production, so perhaps it's fine, but it might be good to at least make this presentable (or remove the menu). That can happen in a followup PR.

ryelle reviewed (commented):
Great update, this is much more discoverable than the little 🔍  tab.

> I noted that if I'd like to get rid of the search suggestion block, I need to use the mouse to click elsewhere on the page instead of just pressing the ESC button. If so, users need to select the filter options first before typing in the search or they must use a mouse. I think in the original UX, the suggestion block can be gotten rid of by pressing ESC.

For what it's worth, using <kbd>ESC</kbd> to close the dropdown is working for me, in both Chrome & Firefox.

I really like this UI for the checked element 💯 

<img width="604" alt="Screen Shot 2022-06-09 at 3 14 27 PM" src="https://user-images.githubusercontent.com/541093/172926640-82a07d5e-0d79-41e6-99a0-f68033c77a87.png">

I did notice the the autocomplete dropdown is now applied to the other searches too, like here in the block editor handbook. Hitting enter (ignoring the suggestions) does direct me to the correct search results page (ex, `/block-editor/?s=block`), so I'd guess the autocomplete JS is a little too overzealous.

<img width="519" alt="Screen Shot 2022-06-09 at 3 34 08 PM" src="https://user-images.githubusercontent.com/541093/172930196-d01112ea-7138-4fbe-bb56-7d8c473ab9f5.png">

StevenDufresne commented:
> I did notice the the autocomplete dropdown is now applied to the other searches too, like here in the block editor handbook. Hitting enter (ignoring the suggestions) does direct me to the correct search results page (ex, `/block-editor/?s=block`), so I'd guess the autocomplete JS is a little too overzealous.

This must be a configuration issue. On the live site, the search gets `class="searchform searchform-handbook"` and the js specifically [excludes `searchform-handbook` search inputs](https://github.com/WordPress/wporg-developer/blob/4fed039f61bf696c45e94692c923bb287ca9f3d8/source/wp-content/themes/wporg-developer/js/autocomplete.js#L14).

StevenDufresne committed:
Add searchbar into a title.

StevenDufresne committed:
Re-add class that was killed by merge.

StevenDufresne commented:
I added code to hide the menu on the homepage/reference, although the menu is still stacked under the search bar when you're looking at a page with the searchbar beside the site-title. Since production doesn't have a menu right now, as @ryelle mentions we can probably just follow it up later.

@renintw I had to squash merge because of conflicts introduced in #40. Do you mind double-checking to make sure I didn't overwrite any changes? I had to manually merge them. 🙏

StevenDufresne committed:
Only show 'code reference' when we are on that page.

outdoor2kode commented:
>> I noted that if I'd like to get rid of the search suggestion block, I need to use the mouse to click elsewhere on the page instead of just pressing the ESC button. If so, users need to select the filter options first before typing in the search or they must use a mouse. I think in the original UX, the suggestion block can be gotten rid of by pressing ESC.

> For what it's worth, using ESC to close the dropdown is working for me, in both Chrome & Firefox.

I should have been more clear. I can use ESC to close dropdown in Chrome as well, but in Chrome it also clears the user input while Firefox doesn't.
It would affect one scenario that if users want to select filter types below, they need to use the mouse to close the dropdown or select the types before inputting.

Note: A possible solution is mentioned in #57

outdoor2kode committed:
Fix button focus style and adjust a11y

outdoor2kode committed:
Add curosr for filter checkbox

outdoor2kode committed:
Fix linting error

outdoor2kode reviewed (approved):
Looks good to me, It's nice!

I've added three commits to adjust styles and a11y and also have rebased to trunk.
Original small style defect (focus state):
![Markup on 2022-06-10 at 22:45:38](https://user-images.githubusercontent.com/18050944/173105017-7ac6327a-f044-4850-93b9-26f4df82999e.png)

ryelle committed:
Remove unused code

The search.js file is not needed anymore, since the tab UI was removed. These CSS selectors are also not used in the markup anymore.

ryelle committed:
Add back the `searchform-handbook` class

ryelle committed:
Update the search placeholder for handbooks

ryelle commented:
> This must be a configuration issue. On the live site, the search gets class="searchform searchform-handbook" and the js specifically [excludes searchform-handbook search inputs](https://github.com/WordPress/wporg-developer/blob/4fed039f61bf696c45e94692c923bb287ca9f3d8/source/wp-content/themes/wporg-developer/js/autocomplete.js#L14).

It wasn't a config issue, this PR was removing the logic that added the class. I [brought it back](https://github.com/WordPress/wporg-developer/pull/75/commits/3a26838a50b940e4e815a6956bdd6e4760fa9e94) & styled the search a little more. I also removed some unused code now that we don't have the drawer-search UI. 

<img width="1001" alt="Screen Shot 2022-06-10 at 2 24 06 PM" src="https://user-images.githubusercontent.com/541093/173127796-71a06200-963b-4f2b-8aac-8b92d50da659.png">
<img width="347" alt="Screen Shot 2022-06-10 at 2 23 55 PM" src="https://user-images.githubusercontent.com/541093/173127801-9b4424cf-ca5d-4e08-bb46-84019d41dee4.png">

I'm going to merge this so it doesn't block other work.

iandunn commented:
Anyone else seeing this in Firefox?

![Screen Shot 2022-06-10 at 1 19 10 PM](https://user-images.githubusercontent.com/484068/173144572-d4342247-46b2-47ba-b6aa-14c630e37168.png)

_( on `trunk` after running `build`, at `1900px` )_

Oddly, the horizontal bar goes away in responsive mode.

iandunn commented:
I included a fix for ^ in #114

StevenDufresne commented:
Thanks for jumping in on this all!
